src/revocation.rs

```diff
@@ -8,9 +8,135 @@ use crate::error::{Error, Result};
 use crate::secure_mount;
 
 use std::path::Path;
+use std::process::{Child, Command, Output, Stdio};
 
 use serde_json::Value;
 
+/// Runs a script with a json value as argument (used for revocation actions)
+pub(crate) fn run_action(
+    dir: &Path,
+    script: &str,
+    json: Value,
+) -> Result<Output> {
+    let raw_json = serde_json::value::to_raw_value(&json)?;
+
+    let child = match Command::new(format!("{}{}", "./", script))
+        .current_dir(dir)
+        .arg(raw_json.get())
```

Comment:
I believe this would pass the data as if it was an argument on a commandline instead of it it going in through standard input. If it was in a shell it would look like:

```
./path/to/script {"hello": ["I", "am", "the", "payload"]}
```

What I think we need is to have the `stdin`. Which would be closer to the following shell example:

```
echo -n '{"hello": ["I", "am", "the", "payload"]}' | ./path/to/script 
```

I've not done this myself in rust before so I searched a bit and [this came up](https://stackoverflow.com/questions/49218599/write-to-child-process-stdin-in-rust) and looks like it's a good example:

Reply:
Ah, I see what you're saying. Thanks for the example, I'll try it that way.